google/cloud/storage/internal/bucket_metadata_parser.cc

```diff
@@ -67,298 +68,444 @@ StatusOr<UniformBucketLevelAccess> ParseUniformBucketLevelAccess(
     }
-    metadata_as_json["iamConfiguration"] = std::move(c);
-  }
-
-  if (!meta.labels().empty()) {
-    nlohmann::json labels_as_json;
-    for (auto const& kv : meta.labels()) {
-      labels_as_json[kv.first] = kv.second;
+    if (c.is_live) {
+      condition["isLive"] = *c.is_live;
     }
-    metadata_as_json["labels"] = std::move(labels_as_json);
-  }
-
-  if (meta.has_lifecycle()) {
-    nlohmann::json rule;
-    for (LifecycleRule const& v : meta.lifecycle().rule) {
-      nlohmann::json condition;
-      auto const& c = v.condition();
-      if (c.age) {
-        condition["age"] = *c.age;
-      }
-      if (c.created_before.has_value()) {
-        condition["createdBefore"] =
-            absl::StrFormat("%04d-%02d-%02d", c.created_before->year(),
-                            c.created_before->month(), c.created_before->day());
-      }
-      if (c.is_live) {
-        condition["isLive"] = *c.is_live;
-      }
-      if (c.matches_storage_class) {
-        condition["matchesStorageClass"] = *c.matches_storage_class;
-      }
-      if (c.num_newer_versions) {
-        condition["numNewerVersions"] = *c.num_newer_versions;
-      }
-      nlohmann::json action{{"type", v.action().type}};
-      if (!v.action().storage_class.empty()) {
-        action["storageClass"] = v.action().storage_class;
-      }
-      rule.emplace_back(nlohmann::json{{"condition", std::move(condition)},
-                                       {"action", std::move(action)}});
+    if (c.matches_storage_class) {
+      condition["matchesStorageClass"] = *c.matches_storage_class;
     }
-    metadata_as_json["lifecycle"] = nlohmann::json{{"rule", std::move(rule)}};
+    if (c.num_newer_versions) {
+      condition["numNewerVersions"] = *c.num_newer_versions;
+    }
+    nlohmann::json action{{"type", v.action().type}};
+    if (!v.action().storage_class.empty()) {
+      action["storageClass"] = v.action().storage_class;
+    }
+    value.emplace_back(nlohmann::json{{"condition", std::move(condition)},
+                                      {"action", std::move(action)}});
   }
+  json["lifecycle"] = nlohmann::json{{"rule", std::move(value)}};
+}
 
-  SetIfNotEmpty(metadata_as_json, "location", meta.location());
+void ToJsonLocation(nlohmann::json& json, BucketMetadata const& meta) {
+  SetIfNotEmpty(json, "location", meta.location());
+}
 
-  SetIfNotEmpty(metadata_as_json, "locationType", meta.location_type());
+void ToJsonLocationType(nlohmann::json& json, BucketMetadata const& meta) {
+  SetIfNotEmpty(json, "locationType", meta.location_type());
+}
 
-  if (meta.has_logging()) {
-    nlohmann::json l;
-    SetIfNotEmpty(l, "logBucket", meta.logging().log_bucket);
-    SetIfNotEmpty(l, "logObjectPrefix", meta.logging().log_object_prefix);
-    metadata_as_json["logging"] = std::move(l);
-  }
+void ToJsonLogging(nlohmann::json& json, BucketMetadata const& meta) {
+  if (!meta.has_logging()) return;
 
```

Comment:
nit: most others don't have this newline. change if you want.

Reply:
Done.